heron/stmgr/src/cpp/manager/stmgr-clientmgr.cpp

```diff
@@ -114,6 +117,8 @@ StMgrClient* StMgrClientMgr::CreateClient(const sp_string& _other_stmgr_id,
   options.set_max_packet_size(config::HeronInternalsConfigReader::Instance()
                                   ->GetHeronStreammgrNetworkOptionsMaximumPacketMb() *
                               1024 * 1024);
+  options.set_high_watermark(high_watermark_);
```

Comment:
1) StMgr() doesn't accept NetworkOptions as parameter, max_packet_size is passed in NetworkOptions for those who accept it.

2) stmgr unit tests want 2 different values in different tests, so you have to provide at least 2 different configs if you really want to read from config. Not to mention other config's.